src/main/kotlin/calculator/domain/Operator.kt

```diff
@@ -0,0 +1,31 @@
+package calculator.domain
+
+import java.lang.IllegalArgumentException
+
+enum class Operator(
+    private val value: String,
+    val operation: (Operand, Operand) -> Operand
+) {
+    PLUS("+", { num1, num2 -> Operand(num1.number + num2.number) }),
+    MINUS("-", { num1, num2 -> Operand(num1.number - num2.number) }),
+    MULTIPLE("*", { num1, num2 -> Operand(num1.number * num2.number) }),
+    DIVIDE(
+        "/",
+        { num1, num2 ->
+            if (num2.number == 0.0)
+                throw IllegalArgumentException("0으로는 나눌 수 없습니다")
+            Operand(num1.number * num2.number)
+        }
+    );
+
+    companion object {
+        fun of(operator: String): Operator {
+            try {
+                return values()
+                    .first { o -> o.value == operator }
+            } catch (e: NoSuchElementException) {
+                throw IllegalArgumentException("올바르지 않은 연산자 입니다")
+            }
+        }
```

Comment:
```suggestion
        return values()
                .find { it.value == operator } ?: throw IllegalArgumentException("올바르지 않은 연산자 입니다")
        }
```
이렇게 하면 더욱 간결하게 나타낼 수 있을 것 같아요!